File: stage1/benchmarks/reference/hello/main.go
package main

import "fmt"

func banner(name string) string { return "hello " + name }
func lucky(base int) int        { return base + 2 }
func isReady(value int) bool    { return value == 42 }

func main() {
	answer := lucky(40)
	ready := isReady(answer)
	if ready {
		fmt.Println(banner("from stage1"))
	} else {
		fmt.Println("stage1 failed")
	}
	fmt.Println(answer)
	fmt.Println(ready)
}

File: stage1/benchmarks/reference/hello/main.rs
fn banner(name: &str) -> String {
    format!("hello {}", name)
}

fn lucky(base: i32) -> i32 {
    base + 2
}

fn is_ready(value: i32) -> bool {
    value == 42
}

fn main() {
    let answer = lucky(40);
    let ready = is_ready(answer);
    if ready {
        println!("{}", banner("from stage1"));
    } else {
        println!("stage1 failed");
    }
    println!("{}", answer);
    println!("{}", ready);
}
